### ISSUE TITLE
Type mismatch

### ISSUE THREAD

**Issue by FinnKeller** [FinnKeller/JAXAtari_Phoenix#7]
There is a type mismatch between step state and reset function. To replicate error :

    game = JaxPhoenix()
    key = jax.random.PRNGKey(0)
    _, state = game.reset(key)

    print("--- Initial State Types ---")
    print(f"  player_x: {state.player_x.dtype}")
    print(f"  enemies_x: {state.enemies_x.dtype}")
    print(f"  projectile_x: {state.projectile_x.dtype}")
    print(f"  enemy_projectile_x: {state.enemy_projectile_x.dtype}")
    print("-" * 20)

    # To provoke the error, we replicate the benchmark's JIT-compiled loop.
    # `jax.lax.scan` requires the state ("carry") to have the exact same
    # data types on every single iteration.
    def loop_body(carry_state, _):
        # The `step` function will change the dtype of `enemy_projectile_x`
        # from int32 to float32 when an enemy fires, causing the error.
        obs, next_state, reward, done, info = game.step(carry_state, Action.NOOP)
        return next_state, obs

    # We JIT the entire scan operation.
    jitted_scan_stepper = jax.jit(
        lambda initial_state: jax.lax.scan(loop_body, initial_state, None, length=500)
    )

    print("\nRunning JIT-compiled scan loop to provoke the type error...")
    try:
        # This call will fail because the dtype of enemy projectiles changes inside the loop.
        final_state, observations = jitted_scan_stepper(state)
        print("Scan finished unexpectedly without error.")
    except Exception as e:
        print("\n--- Successfully Reproduced Benchmark Error ---")
        print(e)
        print("---------------------------------------------")

**Comment by FinnKeller** [comment]
--- Initial State Types ---
  player_x: int32
  enemies_x: float32
  projectile_x: int32
  enemy_projectile_x: int32
--------------------

Running JIT-compiled scan loop to provoke the type error...

--- Successfully Reproduced Benchmark Error ---
scan body function carry input and carry output must have equal types (e.g. shapes and dtypes of arrays), but they differ:

The input carry component carry_state.projectile_x has type int32[] but the corresponding output carry component has type float32[], so the dtypes do not match.

Revise the function so that all output types (e.g. shapes and dtypes) match the corresponding input types.
---------------------------------------------
